server/game/cards/13.3-PoS/UnexpectedGuile.js

```diff
@@ -0,0 +1,29 @@
+const DrawCard = require('../../drawcard.js');
+
+class UnexpectedGuile extends DrawCard {
+    setupCardAbilities(ability) {
+        this.attachmentRestriction(card => card.isShadow() && card.controller === this.controller);
+        this.whileAttached({
+            effect: [
+                ability.effects.modifyStrength(2),
+                ability.effects.addKeyword('Insight')
+            ]
+        });
+
+        this.forcedInterrupt({
+            when: {
+                afterChallenge: event => event.challenge.isParticipating(this.parent)
```

Comment:
afterChallenge corresponds to a winner / loser being chosen. onChallengeFinished is the event we want. This seems to be the first card that uses it!